OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java

```diff
@@ -202,9 +204,32 @@ public AlarmInfo getMostImportantAlarm(MetricsConstants mc, boolean showCameras)
 				kIterator++;
 			}
 		}
+		List<RouteSegmentResult> segments = route.getUpcomingTunnel(500);
```

Comment:
Why length of tunnel everytime is recalculated? Why that AlarmInfo is dynamic??? It means it is not displayed in the waypoints alarm list before you get close to the tunnel